### PR TITLE
Fix many things

### PR DESCRIPTION
**Pull request by stu43005**
This PR will fix many errors displayed in the console and include some improvements

* ~~Don't keep search alive~~ Change search query only when activated
* Make sure browser has `navigator.serviceWorker`
* Links change to use router
* Make sure stats data is available
* Multiview link is no longer need channel name
* Close last video in multiview, will also close the livechat
* Component parameter type error
* Refactor watch toolbar to reuse video card menu
* Reset settings will now reset all stores and add confirm dialog
* ChannelAbout lacks `linkified` directive

That's it :P

### CODE REVIEW

File: src/components/channel/ChannelImg.vue [hunk 1]
```diff
@@ -11,7 +11,7 @@
     :height="size"
     class="d-block"
   />
-  <a v-else :href="`/channel/${channel.id}`" @click.stop>
+  <a v-else :href="`/channel/${channel.id}`" @click.exact.stop.prevent="goToChannel()">
```
Comment (sphinxrave) [inline src/components/channel/ChannelImg.vue:14]: prevents the <a> behavior from kicking off, nice.

File: src/App.vue [hunk 2]
```diff
@@ -196,12 +196,14 @@ export default {
         this.$store.dispatch("loginCheck");
 
         // on update, reresh page and set update notification flag
-        navigator.serviceWorker.addEventListener("controllerchange", () => {
-            if (this.refreshing) return;
-            this.refreshing = true;
-            this.showUpdateDetails = true;
-            window.location.reload();
-        });
+        if (navigator.serviceWorker) {
```
Comment (sphinxrave) [inline src/App.vue:199]: good change

File: src/App.vue [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
     <v-main style="transition: none">
       <PullToRefresh />
-      <keep-alive max="4" exclude="Watch,MugenClips,EditVideo,MultiView,Channel,Playlists,About">
+      <keep-alive max="4" exclude="Watch,MugenClips,EditVideo,MultiView,Channel,Playlists,About,Search">
```
Comment (sphinxrave) [inline src/App.vue:12]: not sure about this one, why's murdering Search a good thing?

File: src/components/common/Stats.vue [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <v-row>
+  <v-row v-if="s">
```
Comment (sphinxrave) [inline src/components/common/Stats.vue:2]: :+1:

File: src/components/watch/WatchToolbar.vue [hunk 5]
```diff
@@ -39,61 +39,21 @@
...
-            {{ $t("component.mainNav.multiview") }}
-          </v-list-item>
-          <v-list-item @click="$store.commit('setReportVideo', video)">
-            <v-icon left :color="video.type === 'clip' ? 'grey' : ''">
-              {{ icons.mdiFlag }}
-            </v-icon>
-            {{ $t("component.reportDialog.title") }}
-          </v-list-item>
-        </v-list>
+        <video-card-menu :video="video" />
```
Comment (sphinxrave) [inline src/components/watch/WatchToolbar.vue:42]: 👀 oh we could've done this? lol

File: src/store/settings.module.js [hunk 6]
```diff
@@ -108,6 +108,8 @@ const mutations = {
     ]),
     resetState(state) {
         Object.assign(state, initialState);
+        localStorage.removeItem("theme");
```
Comment (sphinxrave) [inline src/store/settings.module.js:111]: nice

**Review by sphinxrave — APPROVED** [review]

File: src/components/watch/WatchToolbar.vue (hunk 5 above)
Comment (stu43005) [reply src/components/watch/WatchToolbar.vue:42]: image
Yeah, it looks good
and be consistent with video card

**Pull request merged**